cloud: add doc for Chat2Query (#11959)
pingcap/docs: tidb-cloud/explore-data-with-chat2query.md

---
title: Explore Your Data with AI-Powered Chat2Query (beta)

> summary: Learn how to use Chat2Query, an AI-powered SQL editor, to maximize your data value.

Reviewer: ```suggestion
summary: Learn how to use Chat2Query, an AI-powered SQL editor in the TiDB Cloud console, to maximize your data value.
```

---

# Explore Your Data with AI-Powered Chat2Query (beta)

TiDB Cloud is powered by AI. You can use Chat2Query (beta), an AI-powered SQL editor in the [TiDB Cloud console](https://tidbcloud.com/), to maximize your data value.

> In Chat2Query, you can either simply type `--` and your instructions to let AI generate SQL queries automatically or write SQL queries manually, and then run SQL queries against databases without a terminal. You can find the query results in tables intuitively and check the query logs easily.

Author: ```suggestion
In Chat2Query, you can either simply type `--` followed by your instructions to let AI generate SQL queries automatically or write SQL queries manually, and then run SQL queries against databases without a terminal. You can find the query results in tables intuitively and check the query logs easily.
```

> **Note:**
>
> Chat2Query is only available to [Serverless Tier](/tidb-cloud/select-cluster-tier.md#serverless-tier-beta) clusters.

> ## Usage scenarios
> 
> The recommended usage scenarios of Chat2Query are as follows:

Author: ```suggestion
## Use cases

The recommended use cases of Chat2Query are as follows:
```

> - Use the AI capacity of Chat2Query to help you write complex SQL queries instantly.

Author: ```suggestion
- Use the AI capacity of Chat2Query to help you generate complex SQL queries instantly.
```

- Test out the MySQL compatibility of TiDB quickly.
- Explore TiDB SQL features easily.

## Limitation

> SQL queries generated by AI is not 100% accurate and might still need your further tweak.

Reviewer: ```suggestion
SQL queries generated by AI are not 100% accurate and might still need your further tweak.
```

> SQL queries generated by the AI in Chat2Query are not 100% accurate and might still need your further tweak.

Author: ```suggestion
SQL queries generated by the AI are not 100% accurate and might still need your further tweak.
```

## Access Chat2Query

1. Go to the [**Clusters**](https://tidbcloud.com/console/clusters) page of your project.

    > **Tip:**
    >
    > If you have multiple projects, you can switch to the target project in the left navigation pane of the **Clusters** page.

2. Click your cluster name, and then click **Chat2Query** in the left navigation pane.

## Enable or disable AI to generate SQL queries

> PingCAP takes the privacy and security of users' data as top priority. The AI capacity of Chat2Query only needs to access database schemas to generate SQL queries, not your data itself. For more information, see [Chat2Query Privacy FAQ](https://www.pingcap.com/privacy-policy/privacy-chat2query).

Reviewer: ```suggestion
PingCAP takes the privacy and security of users' data as a top priority. The AI capacity of Chat2Query only needs to access database schemas to generate SQL queries, not your data itself. For more information, see [Chat2Query Privacy FAQ](https://www.pingcap.com/privacy-policy/privacy-chat2query).
```

> For the first-time you access Chat2Query, you will get a dialog about whether to allow PingCAP and OpenAI to use your code snippets to research and improve the services.

Reviewer: ```suggestion
When you access Chat2Query for the first time, you will see a dialog about whether to allow PingCAP and OpenAI to use your code snippets to research and improve the services.
```

- To enable AI to generate SQL queries, select the checkbox and click **Save and Get Started**.
- To disable the AI to generate SQL queries, close this dialog directly.

After the first-time access, you can still change the AI setting as follows:

- To enable AI, click **Enable AI power for data exploration** in the upper-right corner of Chat2Query.
- To disable AI, click <MDSvgIcon name="icon-top-account-settings" /> **Account** in the upper-right corner of the [TiDB Cloud console](https://tidbcloud.com/), click **Account Settings**, click the **Privacy** tab, and then disable the **AI-powered Data Exploration** option.

## Write and run SQL queries

In Chat2Query, you can write and run SQL queries using the pre-built sample dataset or your own dataset.

1. Write SQL queries.

>     - If AI is enabled, simply type `--` and your instructions to let AI generate SQL queries automatically or write SQL queries manually.

Author: ```suggestion
    - If AI is enabled, simply type `--` followed by your instructions to let AI generate SQL queries automatically or write SQL queries manually.
```

        For a SQL query generated by AI, you can accept it by pressing `Tab` and then further edit it if needed, or reject it by pressing `Esc`.

    - If AI is disabled, write SQL queries manually.

2. Run SQL queries.

    <SimpleTab>
    <div label="Mac">

>     For Mac:

Reviewer 2: ```suggestion
    For macOS:
```

>     - If you have only one query in the editor, to run it, press **Command + Enter** or click <svg width="1rem" height="1rem" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M6.70001 20.7756C6.01949 20.3926 6.00029 19.5259 6.00034 19.0422L6.00034 12.1205L6 5.33028C6 4.75247 6.00052 3.92317 6.38613 3.44138C6.83044 2.88625 7.62614 2.98501 7.95335 3.05489C8.05144 3.07584 8.14194 3.12086 8.22438 3.17798L19.2865 10.8426C19.2955 10.8489 19.304 10.8549 19.3126 10.8617C19.4069 10.9362 20 11.4314 20 12.1205C20 12.7913 19.438 13.2784 19.3212 13.3725C19.307 13.3839 19.2983 13.3902 19.2831 13.4002C18.8096 13.7133 8.57995 20.4771 8.10002 20.7756C7.60871 21.0812 7.22013 21.0683 6.70001 20.7756Z" fill="currentColor"></path></svg>**Run**.

Reviewer: On macOS, I can use either **Control + Enter** or **Command + Enter** to run SQL.

Author: According to PM, this is to provide the same instructions as that on the UI.

    - If you have multiple queries in the editor, to run one or several of them sequentially, select the lines of the target queries with your cursor, and then press **Command + Enter** or click **Run**.

    - To run all queries in the editor sequentially, press **Shift + Command + Enter**, or select the lines of all queries with your cursor and click **Run**.

    </div>

    <div label="Windows/Linux">

    For Windows or Linux:

>     - If you have only one query in the editor, to run it, press **Ctrl + Enter** or click <svg width="1rem" height="1rem" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M6.70001 20.7756C6.01949 20.3926 6.00029 19.5259 6.00034 19.0422L6.00034 12.1205L6 5.33028C6 4.75247 6.00052 3.92317 6.38613 3.44138C6.83044 2.88625 7.62614 2.98501 7.95335 3.05489C8.05144 3.07584 8.14194 3.12086 8.22438 3.17798L19.2865 10.8426C19.2955 10.8489 19.304 10.8549 19.3126 10.8617C19.4069 10.9362 20 11.4314 20 12.1205C20 12.7913 19.438 13.2784 19.3212 13.3725C19.307 13.3839 19.2983 13.3902 19.2831 13.4002C18.8096 13.7133 8.57995 20.4771 8.10002 20.7756C7.60871 21.0812 7.22013 21.0683 6.70001 20.7756Z" fill="currentColor"></path></svg>**Run**.

Author: This is to align with UI. 
![img_v2_fe0f2ebb-8bab-4a5f-985e-a8b53d2baecg](https://user-images.githubusercontent.com/79440533/211497117-56599eee-19cf-4130-89f1-3a13157e42ac.jpg)

>     - If you have multiple queries in the editor, to run one or several of them sequentially, select the lines of the target queries with your cursor, and then press **Ctrl + Enter** or click **Run**.

Author: ditto

>     - To run all queries in the editor sequentially, press **Shift + Ctrl + Enter**, or select the lines of all queries with your cursor and click **Run**.

Author: ditto

    </div>
    </SimpleTab>

> After running the queries, you can see the query log and results immediately at the bottom of the page.

Reviewer: ```suggestion
After running the queries, you can see the query logs and results immediately at the bottom of the page.
```

## Manage SQL files

In Chat2Query, you can save your SQL queries in different SQL files and manage SQL files as follows:

- To add a SQL file, click **+** on the **SQL Files** tab.

> - To rename a SQL file, move your cursor on the filename, click **...** next to the filename, and then select **Rename**.

Reviewer: ```suggestion
- To rename an existing SQL file, move your cursor on the filename, click **...** next to the filename, and then select **Rename**.
```

Author: not necessary? It seems that no one will rename a file on the GUI that is not existing.

> - To delete a SQL file, move your cursor on the filename, click **...** next to the filename, and then select **Delete**.

Reviewer: ```suggestion
- To delete an existing SQL file, move your cursor on the filename, click **...** next to the filename, and then select **Delete**.
```

Author: not necessary? It seems that no one will delete a file on the GUI that is not existing.

> - To delete a SQL file, move your cursor on the filename, click **...** next to the filename, and then select **Delete**.

Author: ```suggestion
- To delete a SQL file, move your cursor on the filename, click **...** next to the filename, and then select **Delete**. Note that when there is only one SQL file on the **SQL Files** tab, you cannot delete it.
```

## Manage Chat2Query settings

By default, Chat2Query limits the maximum number of rows in query results to 500 and does not show system database schemas on the **Schemas** tab.

To change the settings, take the following steps:

> 1. In the upper-right corner of Chat2Query, click **...** and select **Chat2Query settings**.

Reviewer: ```suggestion
1. In the upper-right corner of Chat2Query, click **...** and select **Settings**.
```
<img width="193" alt="image" src="https://user-images.githubusercontent.com/60599231/211495587-63d763f1-f9ed-4845-941d-cdb92a64ce85.png">

Author: Good catch. It was newly updated today.
